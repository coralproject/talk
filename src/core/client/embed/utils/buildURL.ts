export default function buildURL({
  protocol = window.location.protocol,
  hostname = window.location.hostname,
  port = window.location.port,
  pathname = window.location.pathname,
  search = window.location.search,
  hash = window.location.hash,
} = {}) {
  if (search && search[0] !== "?") {
    search = `?${search}`;
  } else if (search === "?") {
    search = "";
  }
  return `${protocol}//${hostname}${
    port ? `:${port}` : ""
  }${pathname}${search}${hash}`;
}
