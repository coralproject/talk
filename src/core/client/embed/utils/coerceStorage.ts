import { createInMemoryStorage } from "coral-framework/lib/storage";

function coerceStorage(type: "localStorage" | "sessionStorage"): Storage {
  let storage: Storage;
  try {
    storage = window[type];
  } catch {
    // eslint-disable-next-line no-console
    console.debug(`${type} not available, use in-memory replacement`);

    storage = createInMemoryStorage();
  }

  return storage;
}

export default coerceStorage;
