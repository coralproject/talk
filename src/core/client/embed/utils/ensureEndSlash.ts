export default function ensureEndSlash(p: string) {
  return p.match(/\/$/) ? p : `${p}/`;
}
