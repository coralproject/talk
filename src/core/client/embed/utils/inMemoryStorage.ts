class InMemoryStorage implements Storage {
  private readonly data = new Map<string, string>();

  public get length() {
    return this.data.size;
  }

  public clear() {
    this.data.clear();
  }

  public key(n: number) {
    const keys = [...this.data.keys()];
    if (keys.length < n || n < 0) {
      return null;
    }

    return keys[n];
  }

  public getItem(key: string): string | null {
    return this.data.get(key) || null;
  }

  public removeItem(key: string): void {
    this.data.delete(key);
  }

  public setItem(key: string, value: string): void {
    this.data.set(key, value);
  }
}

export default InMemoryStorage;
