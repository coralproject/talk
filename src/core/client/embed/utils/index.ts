export { default as buildURL } from "./buildURL";
export { default as ensureEndSlash } from "./ensureEndSlash";
export { default as ensureNoEndSlash } from "./ensureNoEndSlash";
export { default as startsWith } from "./startsWith";
export { default as prefixStorage } from "./prefixStorage";
export { default as parseHashQuery } from "./parseHashQuery";
