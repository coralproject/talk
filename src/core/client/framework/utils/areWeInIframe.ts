/**
 * Returns true if we are in an iframe.
 */
export default function areWeInIframe() {
  try {
    return window.self !== window.top;
  } catch (e) {
    return true;
  }
}
