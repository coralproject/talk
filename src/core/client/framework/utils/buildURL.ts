/* eslint-disable no-restricted-globals */
import startsWith from "coral-common/utils/startsWith";

export default function buildURL({
  protocol = window.location.protocol,
  hostname = window.location.hostname,
  port = window.location.port,
  pathname = window.location.pathname,
  search = window.location.search,
  hash = window.location.hash,
} = {}) {
  if (search && !startsWith(search, "?")) {
    search = `?${search}`;
  } else if (search === "?") {
    search = "";
  }
  if (!startsWith(pathname, "/")) {
    pathname = "/" + pathname;
  }
  return `${protocol}//${hostname}${
    port ? `:${port}` : ""
  }${pathname}${search}${hash}`;
}
