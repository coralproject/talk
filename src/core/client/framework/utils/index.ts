export { default as buildURL } from "./buildURL";
export { default as parseURL } from "./parseURL";
export { default as modifyQuery } from "./modifyQuery";
