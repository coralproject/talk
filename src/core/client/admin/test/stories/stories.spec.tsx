import { noop } from "lodash";

import { pureMerge } from "coral-common/utils";
import {
  GQLResolver,
  GQLSTORY_STATUS,
  MutationToCloseStoryResolver,
  MutationToOpenStoryResolver,
} from "coral-framework/schema";
import {
  act,
  createMutationResolverStub,
  createResolversStub,
  CreateTestRendererParams,
  findParentWithType,
  replaceHistoryLocation,
  wait,
  waitForElement,
  waitUntilThrow,
  within,
} from "coral-framework/testHelpers";

import create from "../create";
import {
  emptyStories,
  settings,
  site,
  siteConnection,
  stories,
  storyConnection,
  users,
} from "../fixtures";

const viewer = users.admins[0];

beforeEach(async () => {
  replaceHistoryLocation("http://localhost/admin/stories");
});

async function createTestRenderer(
  params: CreateTestRendererParams<GQLResolver> = {}
) {
  const { testRenderer, context } = create({
    ...params,
    resolvers: pureMerge(
      createResolversStub<GQLResolver>({
        Query: {
          settings: () => settings,
          viewer: () => viewer,
          site: () => site,
          sites: () => siteConnection,
          stories: ({ variables }) => {
            expectAndFail(variables.status).toBeFalsy();
            return storyConnection;
          },
        },
      }),
      params.resolvers
    ),
    initLocalState: (localRecord, source, environment) => {
      if (params.initLocalState) {
        params.initLocalState(localRecord, source, environment);
      }
    },
  });

  return await act(async () => {
    const container = await waitForElement(() =>
      within(testRenderer.root).getByTestID("stories-container")
    );

    return { testRenderer, container, context };
  });
}

it("renders stories", async () => {
  const { container } = await createTestRenderer();
  await act(async () => {
    await wait(() => {
      expect(within(container).toJSON()).toMatchSnapshot();
    });
  });
});

it("renders empty stories", async () => {
  const { container } = await createTestRenderer({
    resolvers: createResolversStub<GQLResolver>({
      Query: {
        users: () => emptyStories,
      },
    }),
  });

  await act(async () => {
    await wait(() => {
      expect(within(container).toJSON()).toMatchSnapshot();
    });
  });
});

it("goes to moderation when clicking on title", async () => {
  const {
    container,
    context: { transitionControl },
  } = await createTestRenderer();

  // Prevent router transitions.
  transitionControl.allowTransition = false;

  const story = storyConnection.edges[0].node;
  act(() => {
    within(container)
      .getByText(story.metadata!.title!)
      .props.onClick({ button: 0, preventDefault: noop });
  });

  // Expect a routing request was made to the right url.
  await act(async () => {
    await wait(() => {
      expect(transitionControl.history[0].pathname).toBe(
        `/admin/moderate/stories/${story.id}`
      );
    });
  });
});

it("filter by status", async () => {
  const { container } = await createTestRenderer({
    resolvers: createResolversStub<GQLResolver>({
      Query: {
        stories: ({ variables, callCount }) => {
          switch (callCount) {
            case 0:
              return storyConnection;
            default:
              expectAndFail(variables.status).toBe(GQLSTORY_STATUS.CLOSED);
              return emptyStories;
          }
        },
      },
    }),
  });

  const selectField = within(container).getByLabelText("Search by status");
  const closedOption = within(selectField).getByText("Closed Stories");

  act(() => {
    selectField.props.onChange({
      target: { value: closedOption.props.value.toString() },
    });
  });

  await act(async () => {
    await waitForElement(() =>
      within(container).getByText("could not find any", { exact: false })
    );
  });
});

it("change story status", async () => {
  const story = stories[1];
  const openStory = createMutationResolverStub<MutationToOpenStoryResolver>(
    ({ variables }) => {
      expectAndFail(variables).toMatchObject({
        id: story.id,
      });
      const storyRecord = pureMerge(story, {
        status: GQLSTORY_STATUS.OPEN,
        createdAt: false,
        isClosed: false,
      });
      return {
        story: storyRecord,
      };
    }
  );

  const closeStory = createMutationResolverStub<MutationToCloseStoryResolver>(
    ({ variables }) => {
      expectAndFail(variables).toMatchObject({
        id: story.id,
      });
      const storyRecord = pureMerge(story, {
        status: GQLSTORY_STATUS.CLOSED,
        createdAt: "2018-11-29T16:01:51.897Z",
        isClosed: true,
      });
      return {
        story: storyRecord,
      };
    }
  );

  const { container } = await createTestRenderer({
    resolvers: {
      Mutation: { openStory, closeStory },
    },
  });

  const storyRow = within(container).getByText(story.metadata!.title!, {
    selector: "tr",
  });

  const changeStatusButton = within(storyRow).getByLabelText("Change status");
  const popup = within(storyRow).getByLabelText(
    "A dropdown to change the story status"
  );

  /** CLOSE STORY */
  act(() => {
    changeStatusButton.props.onClick();
  });
  act(() => {
    within(popup)
      .getByText("Closed", { selector: "button" })
      .props.onClick();
  });

  within(storyRow).getByText("Closed");
  expect(closeStory.called).toBe(true);

  /** OPEN STORY */
  act(() => {
    changeStatusButton.props.onClick();
  });

  act(() => {
    within(popup)
      .getByText("Open", { selector: "button" })
      .props.onClick();
  });

  within(storyRow).getByText("Open");
  expect(openStory.called).toBe(true);
});

it("load more", async () => {
  const { container } = await createTestRenderer({
    resolvers: createResolversStub<GQLResolver>({
      Query: {
        stories: ({ callCount }) => {
          switch (callCount) {
            case 0:
              return {
                edges: [
                  { node: stories[0], cursor: stories[0].createdAt },
                  { node: stories[1], cursor: stories[1].createdAt },
                ],
                pageInfo: {
                  endCursor: stories[1].createdAt,
                  hasNextPage: true,
                },
              };
            default:
              return {
                edges: [{ node: stories[2], cursor: stories[2].createdAt }],
                pageInfo: {
                  endCursor: stories[2].createdAt,
                  hasNextPage: false,
                },
              };
          }
        },
      },
    }),
  });
  const loadMore = within(container).getByText("Load More");
  act(() => {
    loadMore.props.onClick();
  });

  await act(async () => {
    // Wait for load more to disappear.
    await waitUntilThrow(() => within(container).getByText("Load More"));
  });

  await act(async () => {
    await wait(() => {
      // Make sure third user was added.
      within(container).getByText(stories[2].metadata!.title!);
    });
  });
});

it("filter by search", async () => {
  const { container } = await createTestRenderer({
    resolvers: createResolversStub<GQLResolver>({
      Query: {
        stories: ({ variables, callCount }) => {
          switch (callCount) {
            case 0:
              return storyConnection;
            default:
              expectAndFail(variables.query).toBe("search");
              return emptyStories;
          }
        },
      },
    }),
  });

  const searchField = within(container).getByLabelText(
    "Search by story title",
    { exact: false }
  );
  const form = findParentWithType(searchField, "form")!;

  act(() =>
    searchField.props.onChange({
      target: { value: "search" },
    })
  );
  act(() => {
    form.props.onSubmit();
  });

  await act(async () => {
    await waitForElement(() =>
      within(container).getByText("could not find any", { exact: false })
    );
  });
});

it("use searchFilter from url", async () => {
  const searchFilter = "CandyMountain";
  replaceHistoryLocation(`http://localhost/admin/stories?q=${searchFilter}`);
  const { container } = await createTestRenderer({
    resolvers: createResolversStub<GQLResolver>({
      Query: {
        stories: ({ variables }) => {
          expectAndFail(variables.query).toBe(searchFilter);
          return emptyStories;
        },
      },
    }),
  });

  const searchField = within(container).getByLabelText(
    "Search by story title",
    { exact: false }
  );

  await act(async () => {
    await wait(() => {
      expect(searchField.props.value).toBe(searchFilter);
    });
  });
});
