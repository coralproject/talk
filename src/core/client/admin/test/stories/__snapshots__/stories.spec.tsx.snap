// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty stories 1`] = `
<div
  className="MainLayout-root Stories-root"
  data-testid="stories-container"
>
  <div
    className="Box-root HorizontalGutter-root HorizontalGutter-double"
  >
    <div
      className="Box-root Flex-root Flex-flex Flex-doubleItemGutter gutter"
    >
      <div
        className="Box-root Flex-root Flex-flex Flex-doubleItemGutter gutter"
      >
        <fieldset
          className="FieldSet-root"
        >
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-spacing-2"
          >
            <label
              className="Label-root"
            >
              Search
            </label>
            <form
              autoComplete="off"
              id="configure-form"
              onSubmit={[Function]}
            >
              <div
                className="TextField-root StoryTableFilter-textField"
              >
                <input
                  aria-label="Search by story title or author"
                  className="TextField-input TextField-colorDark TextField-seamlessAdornment"
                  name="search"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search by story title or author..."
                  type="text"
                  value=""
                />
                <div
                  className="TextField-adornment"
                >
                  <button
                    aria-label="Search"
                    className="BaseButton-root Button-root Button-sizeRegular Button-colorDark Button-variantRegular Button-uppercase Button-adornmentRight StoryTableFilter-adornment"
                    data-color="dark"
                    data-variant="regular"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    type="submit"
                  >
                    <i
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      search
                    </i>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </fieldset>
        <fieldset
          className="FieldSet-root"
        >
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-spacing-2"
          >
            <label
              className="Label-root"
            >
              Status
            </label>
            <span
              className="SelectField-root"
            >
              <select
                aria-label="Search by status"
                className="SelectField-select"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  All Stories
                </option>
                <option
                  value="OPEN"
                >
                  Open Stories
                </option>
                <option
                  value="CLOSED"
                >
                  Closed Stories
                </option>
              </select>
              <span
                aria-hidden={true}
                className="SelectField-afterWrapper"
              >
                <i
                  aria-hidden="true"
                  className="Icon-root Icon-sm"
                >
                  expand_more
                </i>
              </span>
            </span>
          </div>
        </fieldset>
      </div>
      <fieldset
        className="FieldSet-root"
      >
        <div
          className="Box-root HorizontalGutter-root HorizontalGutter-spacing-2"
        >
          <label
            className="Label-root"
          >
            Site
          </label>
          <div
            className="Popover-root"
          >
            <button
              className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantFlat Button-adornmentLeft PaginatedSelect-button SiteFilter-root"
              data-color="mono"
              data-variant="flat"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="button"
            >
              <div
                className="Box-root Flex-root PaginatedSelect-wrapper Flex-flex Flex-alignCenter"
              >
                <span
                  className="SiteFilter-buttonText"
                >
                  All sites
                </span>
              </div>
              <i
                aria-hidden="true"
                className="Icon-root Icon-sm ButtonIcon-root PaginatedSelect-buttonIconRight"
              >
                keyboard_arrow_down
              </i>
            </button>
            <div
              aria-hidden={true}
              aria-labelledby="-ariainfo"
              id=""
              role="dialog"
            />
          </div>
        </div>
      </fieldset>
    </div>
    <div
      className="Box-root HorizontalGutter-root HorizontalGutter-double"
    >
      <table
        className="Table-root Table-fullWidth"
      >
        <thead
          className="TableHead-root"
        >
          <tr
            className="TableRow-root"
          >
            <th
              className="TableCell-root StoryTable-titleColumn TableCell-header"
            >
              <span>
                Title
              </span>
               
              <span
                className="StoryTable-clickToModerate"
              >
                (
                <span>
                  Click title to moderate story
                </span>
                )
              </span>
            </th>
            <th
              className="TableCell-root StoryTable-authorColumn TableCell-header"
            >
              Author
            </th>
            <th
              className="TableCell-root StoryTable-siteColumn TableCell-header"
            >
              Site
            </th>
            <th
              className="TableCell-root StoryTable-publishDateColumn TableCell-header"
            >
              Publish Date
            </th>
            <th
              className="TableCell-root StoryTable-statusColumn TableCell-header"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody
          className="TableBody-root"
        >
          <tr
            className="TableRow-root TableRow-body"
          >
            <td
              className="TableCell-root StoryRow-titleColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/stories/story-1"
                onClick={[Function]}
              >
                Finally a Cure for Cancer
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-authorColumn TableCell-body"
            >
              Vin Hoa
            </td>
            <td
              className="TableCell-root StoryRow-siteColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/sites/site-1"
                onClick={[Function]}
              >
                Test Site
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-publishDateColumn TableCell-body"
            >
              11/29/2018, 4:01 PM
            </td>
            <td
              className="TableCell-root StoryRow-statusColumn TableCell-body"
            >
              <div
                className="Popover-root"
              >
                <button
                  aria-label="Change status"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText StoryStatusChange-button"
                  data-color="mono"
                  data-variant="text"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <span
                    className="StoryStatusText-root StoryStatusText-open"
                  >
                    Open
                  </span>
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-lg ButtonIcon-root"
                  >
                    arrow_drop_down
                  </i>
                </button>
                <div
                  aria-hidden={true}
                  aria-labelledby="stories-statusChange-ariainfo"
                  id="stories-statusChange"
                  role="dialog"
                >
                  <div
                    className="AriaInfo-root"
                    id="stories-statusChange-ariainfo"
                  >
                    A dropdown to change the story status
                  </div>
                </div>
              </div>
            </td>
          </tr>
          <tr
            className="TableRow-root TableRow-body"
          >
            <td
              className="TableCell-root StoryRow-titleColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/stories/story-2"
                onClick={[Function]}
              >
                First Colony on Mars
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-authorColumn TableCell-body"
            >
              Linh Nguyen
            </td>
            <td
              className="TableCell-root StoryRow-siteColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/sites/site-2"
                onClick={[Function]}
              >
                Second Site
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-publishDateColumn TableCell-body"
            >
              11/29/2018, 4:01 PM
            </td>
            <td
              className="TableCell-root StoryRow-statusColumn TableCell-body"
            >
              <div
                className="Popover-root"
              >
                <button
                  aria-label="Change status"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText StoryStatusChange-button"
                  data-color="mono"
                  data-variant="text"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <span
                    className="StoryStatusText-root StoryStatusText-open"
                  >
                    Open
                  </span>
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-lg ButtonIcon-root"
                  >
                    arrow_drop_down
                  </i>
                </button>
                <div
                  aria-hidden={true}
                  aria-labelledby="stories-statusChange-ariainfo"
                  id="stories-statusChange"
                  role="dialog"
                >
                  <div
                    className="AriaInfo-root"
                    id="stories-statusChange-ariainfo"
                  >
                    A dropdown to change the story status
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`renders stories 1`] = `
<div
  className="MainLayout-root Stories-root"
  data-testid="stories-container"
>
  <div
    className="Box-root HorizontalGutter-root HorizontalGutter-double"
  >
    <div
      className="Box-root Flex-root Flex-flex Flex-doubleItemGutter gutter"
    >
      <div
        className="Box-root Flex-root Flex-flex Flex-doubleItemGutter gutter"
      >
        <fieldset
          className="FieldSet-root"
        >
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-spacing-2"
          >
            <label
              className="Label-root"
            >
              Search
            </label>
            <form
              autoComplete="off"
              id="configure-form"
              onSubmit={[Function]}
            >
              <div
                className="TextField-root StoryTableFilter-textField"
              >
                <input
                  aria-label="Search by story title or author"
                  className="TextField-input TextField-colorDark TextField-seamlessAdornment"
                  name="search"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search by story title or author..."
                  type="text"
                  value=""
                />
                <div
                  className="TextField-adornment"
                >
                  <button
                    aria-label="Search"
                    className="BaseButton-root Button-root Button-sizeRegular Button-colorDark Button-variantRegular Button-uppercase Button-adornmentRight StoryTableFilter-adornment"
                    data-color="dark"
                    data-variant="regular"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    type="submit"
                  >
                    <i
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      search
                    </i>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </fieldset>
        <fieldset
          className="FieldSet-root"
        >
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-spacing-2"
          >
            <label
              className="Label-root"
            >
              Status
            </label>
            <span
              className="SelectField-root"
            >
              <select
                aria-label="Search by status"
                className="SelectField-select"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  All Stories
                </option>
                <option
                  value="OPEN"
                >
                  Open Stories
                </option>
                <option
                  value="CLOSED"
                >
                  Closed Stories
                </option>
              </select>
              <span
                aria-hidden={true}
                className="SelectField-afterWrapper"
              >
                <i
                  aria-hidden="true"
                  className="Icon-root Icon-sm"
                >
                  expand_more
                </i>
              </span>
            </span>
          </div>
        </fieldset>
      </div>
      <fieldset
        className="FieldSet-root"
      >
        <div
          className="Box-root HorizontalGutter-root HorizontalGutter-spacing-2"
        >
          <label
            className="Label-root"
          >
            Site
          </label>
          <div
            className="Popover-root"
          >
            <button
              className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantFlat Button-adornmentLeft PaginatedSelect-button SiteFilter-root"
              data-color="mono"
              data-variant="flat"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="button"
            >
              <div
                className="Box-root Flex-root PaginatedSelect-wrapper Flex-flex Flex-alignCenter"
              >
                <span
                  className="SiteFilter-buttonText"
                >
                  All sites
                </span>
              </div>
              <i
                aria-hidden="true"
                className="Icon-root Icon-sm ButtonIcon-root PaginatedSelect-buttonIconRight"
              >
                keyboard_arrow_down
              </i>
            </button>
            <div
              aria-hidden={true}
              aria-labelledby="-ariainfo"
              id=""
              role="dialog"
            />
          </div>
        </div>
      </fieldset>
    </div>
    <div
      className="Box-root HorizontalGutter-root HorizontalGutter-double"
    >
      <table
        className="Table-root Table-fullWidth"
      >
        <thead
          className="TableHead-root"
        >
          <tr
            className="TableRow-root"
          >
            <th
              className="TableCell-root StoryTable-titleColumn TableCell-header"
            >
              <span>
                Title
              </span>
               
              <span
                className="StoryTable-clickToModerate"
              >
                (
                <span>
                  Click title to moderate story
                </span>
                )
              </span>
            </th>
            <th
              className="TableCell-root StoryTable-authorColumn TableCell-header"
            >
              Author
            </th>
            <th
              className="TableCell-root StoryTable-siteColumn TableCell-header"
            >
              Site
            </th>
            <th
              className="TableCell-root StoryTable-publishDateColumn TableCell-header"
            >
              Publish Date
            </th>
            <th
              className="TableCell-root StoryTable-statusColumn TableCell-header"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody
          className="TableBody-root"
        >
          <tr
            className="TableRow-root TableRow-body"
          >
            <td
              className="TableCell-root StoryRow-titleColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/stories/story-1"
                onClick={[Function]}
              >
                Finally a Cure for Cancer
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-authorColumn TableCell-body"
            >
              Vin Hoa
            </td>
            <td
              className="TableCell-root StoryRow-siteColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/sites/site-1"
                onClick={[Function]}
              >
                Test Site
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-publishDateColumn TableCell-body"
            >
              11/29/2018, 4:01 PM
            </td>
            <td
              className="TableCell-root StoryRow-statusColumn TableCell-body"
            >
              <div
                className="Popover-root"
              >
                <button
                  aria-label="Change status"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText StoryStatusChange-button"
                  data-color="mono"
                  data-variant="text"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <span
                    className="StoryStatusText-root StoryStatusText-open"
                  >
                    Open
                  </span>
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-lg ButtonIcon-root"
                  >
                    arrow_drop_down
                  </i>
                </button>
                <div
                  aria-hidden={true}
                  aria-labelledby="stories-statusChange-ariainfo"
                  id="stories-statusChange"
                  role="dialog"
                >
                  <div
                    className="AriaInfo-root"
                    id="stories-statusChange-ariainfo"
                  >
                    A dropdown to change the story status
                  </div>
                </div>
              </div>
            </td>
          </tr>
          <tr
            className="TableRow-root TableRow-body"
          >
            <td
              className="TableCell-root StoryRow-titleColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/stories/story-2"
                onClick={[Function]}
              >
                First Colony on Mars
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-authorColumn TableCell-body"
            >
              Linh Nguyen
            </td>
            <td
              className="TableCell-root StoryRow-siteColumn TableCell-body"
            >
              <a
                className="TextLink-root"
                href="/admin/moderate/sites/site-2"
                onClick={[Function]}
              >
                Second Site
              </a>
            </td>
            <td
              className="TableCell-root StoryRow-publishDateColumn TableCell-body"
            >
              11/29/2018, 4:01 PM
            </td>
            <td
              className="TableCell-root StoryRow-statusColumn TableCell-body"
            >
              <div
                className="Popover-root"
              >
                <button
                  aria-label="Change status"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText StoryStatusChange-button"
                  data-color="mono"
                  data-variant="text"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <span
                    className="StoryStatusText-root StoryStatusText-open"
                  >
                    Open
                  </span>
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-lg ButtonIcon-root"
                  >
                    arrow_drop_down
                  </i>
                </button>
                <div
                  aria-hidden={true}
                  aria-labelledby="stories-statusChange-ariainfo"
                  id="stories-statusChange"
                  role="dialog"
                >
                  <div
                    className="AriaInfo-root"
                    id="stories-statusChange-ariainfo"
                  >
                    A dropdown to change the story status
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
